snooty/diagnostics.py

```diff
@@ -696,3 +696,19 @@ def __init__(
             start,
             end,
         )
+
+
+class IconMustBeDefined(Diagnostic):
+    severity = Diagnostic.Level.error
+
+    def __init__(
+        self,
+        icon_role: str,
+        start: Union[int, Tuple[int, int]],
+        end: Union[None, int, Tuple[int, int]] = None,
+    ) -> None:
+        super().__init__(
+            f"Icon roles must be defined in .icon_names, supplied icon role is not defined: {icon_role}",
```

Comment:
This message is a little implementation-specific and has a lot of stuff that writers may not care about. I might softly suggest
```
f"The Icon {icon_role} does not exist"
```
despite the lesser amount of information communicated.